compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt

```diff
@@ -479,12 +479,21 @@ class ComposeScene internal constructor(
      */
     val contentSize: IntSize
         get() {
-            check(!isClosed) { "ComposeScene is closed" }
             val mainOwner = mainOwner ?: return IntSize.Zero
-            mainOwner.measureAndLayout()
-            return mainOwner.contentSize
+            return contentSizeInConstraints(mainOwner.constraints)
         }
 
+    /**
+     * Provides [ComposeScene] content size in given [constraints]
+     *
+     * This function doesn't change current [ComposeScene] scene constraints.
+     */
+    @ExperimentalComposeUiApi
+    fun contentSizeInConstraints(constraints: Constraints): IntSize {
```

Comment:
After we apply infiinity contraints, what do you think, if we:
1. remove `contentSizeInConstraints`
2. place this code into `contentSize`
3. cache `_contentSize` and reset it in `neadLayout`

Reply:
contentSize sets infinity constraints itself. it was added fort that in the first place, not to know the content size in the current constraints

Reply:
Discussed that contentSize should return size in infinity contraints by design. This will be a breaking change, but if someone used this API before, most likely it was used inproperly

Reply:
Updated PR, please check the new approach

Reply:
I would like to avoid redundant caching and complicate invalidation, so I just provided calculateContentSize function